rfcs/000-resource-notices.md

```diff
@@ -0,0 +1,94 @@
+
+## Reference-level Explanation (aka Technical Details)
+
+The design we propose is based on the following:
+
+- Using our current frontend configuration files (config-utils.ts, config-default.ts, config-custom.ts, and config-types.ts)
+- Allowing a three-level "severity" schema for the alerts (info, warning, and alert)
+- Allowing its use in "tables" and "dashboards" (we could add it to people as well, but we'll need designs for it)
+
+What we are not supporting for now:
+
+- Multiple notifications per resource
+- Loading of external files (like a JSON)
+- Application of a single notice to multiple resources
+
+In the implementation, we'll check the notices object whenever we visit the resource detail page, showing any notification if present.
+
+It is not a breaking change, there won't be any costs for older versions to support this.
+
+## Drawbacks
+
+- This implementation could not be scalable, as maybe users want to have too many notifications. It doesn't mean this work won't be a good start, but it could be not enough. TBD
+- With this approach, to add/update/remove a notification, admins would need to make a new deployment of the application. A better approach would be if we use a UI, so a proper feature flag system will be much better for this.
+- Notices are on a 1:1 basis. If admins wanted to deprecate a whole schema or dashboard product, they would need to create a notice for each table/dashboard.
+
+## Alternatives
+
+- Integrating a proper feature flag system and feed the configuration from it. This would be ideal not just for this feature but for many others. It is also more complex and costly.
+- Using a JSON file to feed the notices was also considered. Given that we are not happy with the current configuration options (JS config + Python configuration), adding yet another one didn't feel like a good move.
+- If we don't do anything, we won't be able to show this information in a highlighted way (we would probably add this as a programmatic description, making it less visible)
+
+## Prior art
+
+Nothing that we know of
+
+## Unresolved questions
+
+We think this proposal is a good starting point. It might be lacking in different ways, but we don't know yet in which ways. That's why we think this would be only the first iteration.
+
+## Future possibilities
+
+Out of scope and pending feedback for this stage will be:
+
+- A different way of storing the notices (JSON file, as part of the graph, UI fed)
+- Multiple notifications at once and how we would limit those
+- Notifications in the User pages
+
+A proposed roadmap is:
+
+1. MVP: 1 non-dismissable notice per resource | notice data stored in ConfigUtils
+2. v1: up to 3 dismissible notices per resource shown, ranked by recency | notice data stored in neo4j
+3. v2: form for creating notices | access control so only owners or admins can add or remove notices on their resources from the UI
```

Comment:
maybe we should also consider api for dynamically generating notices. I can imagine situation where some ETL pipeline validates data quality and based on it sends a notice `this table has not passed latest quality check`

Reply:
Good call, adding it!